models/Conversation.js

```diff
@@ -68,6 +71,41 @@ ConversationSchema.method({
       this.sendRandomMessage();
     }, interval);
   },
+
+  generateToken() {
+    return uuidv4();
+  },
+
+  sendVerificationEmail(cb) {
+    const subject = 'Verify your email';
+    const from = 'ssing128@gmail.com';
+    const body = `http://localhost:8080/#/conversations/${this._id}/verify?token=${this.verifyToken}`;
+    const mail = new Mail(
+      this.to_email,
+      from,
+      subject,
+      body,
+      process.env.CONFIRMATION_EMAIL_TEMPLATE_ID,
+    );
+
+    mail.sendEmail((err) => {
+      if (err) {
+        cb(err);
+      }
+    });
+  },
+
+  verifyTokenEmail(queryToken, cb) {
+    if (this.verifyToken === queryToken) {
+      this.verifyToken = '';
+      this.save((err) => {
+        cb(err, this);
+      });
+    } else if (this.verifyToken !== queryToken) {
+      const err = { tokenMatch: false };
```

Comment:
Error should probably be an error object. `const err = new Error('Verification token does not match.')`